### DISCUSSION TITLE
Calling executables in isolated environments like Flatpak or Docker

### DISCUSSION THREAD

**Discussion by Antyos** [microsoft/vscode-discussions#377] [Extension Development QnA]
I recently received an issue on one on my extensions which offers language support for the CAD software OpenSCAD. The issue (https://github.com/Antyos/vscode-openscad/issues/36) details how OpenSCAD installed through Flatpak can't be run. I wanted to see if any other extension authors have had similar issues in their extensions, and what they've done to resolve them. 

My extension first checks if the path specified by the extension configuration is a valid command before allowing execution to continue. This approach has worked well up until now where the command is run through a package manager like Flatpak. Of course the check fails because a call to a Flatpak package isn't a command that "exists" in the traditional sense.

As detailed in the issue, there are a few options I can think of:
| Idea | Problems |
|:-----|:----------|
| 1. Hard code support for Flatpak. | There are likely other package managers similar to Flatpak and coding them all in manually is not feasible. For instance, there is also a Docker image for OpenSCAD. I don't know if can support a gui, but I think limited functionality should still be supported. |
| 2. Try the specified command with a known parameter like `--version`, and only raise an error if it fails. | This gets rid of the safety check for the command existing in the first place, and generally feels a bit less "safe". |
| 3. Parse the configuration and pull out the first token as the command and converting the rest to arguments. | Setting aside needing to skip spaces in quotes (I'm sure there are plenty of ways to do that), if the command is to a package manager, there isn't a direct way to tell if OpenSCAD is installed without hard coding how each package manager does it. Also, if possible, I would still want users to be able to specify a path with spaces without the need for quotes in the configuration. |

What are your thoughts? Maybe some combination of 2 and 3?

**Comment by burkeholland** [comment]
It's been a while on this question with no answer. Closing this for now, but please feel free to re-open if this is still an issue.

**Reply by Antyos** [reply]
I eventually figured it out, I went with a variation of 2. If anyone is interested, they can check out [Antyos/vscode-openscad#43](https://github.com/Antyos/vscode-openscad/pull/43)

We're good to close 👍
